manager/controlapi/node.go

```diff
@@ -273,7 +273,8 @@ func (s *Server) RemoveNode(ctx context.Context, request *api.RemoveNodeRequest)
 		if node == nil {
 			return grpc.Errorf(codes.NotFound, "node %s not found", request.NodeID)
 		}
-		if node.Spec.Role == api.NodeRoleManager {
+		member := s.raft.GetMemberByNodeID(request.NodeID)
```

Comment:
Is this check even useful? We do a double check here, I'm not sure what was the initial intent to test both on the fact that the node was in the memberlist and that it had the role set to Manager.

Seems to me like this can be refactored by removing the below portion and only rely on the first check.